File: java/src/com/esielkar/builder/Director.java
package com.esielkar.builder;

public class Director {
    private Builder builder;

    public Director(Builder builder) {
        this.builder = builder;
    }

    public void changeBuilder(Builder builder) {
        this.builder = builder;
    }

    public void make(int type) {
        switch (type) {
            case 1:
                builder.buildStepA();
                break;
            case 2:
                builder.buildStepA().buildStepB();
                break;
            default:
                builder.buildStepA().buildStepB().buildStepC();

        }
    }
}

File: dart/design_patterns/bin/builder/director.dart
import 'builder.dart';

class Director {
  Builder _builder;

  Director(Builder builder) : _builder = builder;

  set builder(Builder builder) {
    _builder = builder;
  }

  void make(int type) {
    switch (type) {
      case 1:
        _builder.buildStepA();
        break;
      case 2:
        _builder
          ..buildStepA()
          ..buildStepB();
        break;
      default:
        _builder
          ..buildStepA()
          ..buildStepB()
          ..buildStepC();
    }
  }
}
